pkg/kn/commands/service/configuration_edit_flags.go

```diff
@@ -160,10 +162,22 @@ func (p *ConfigurationEditFlags) addSharedFlags(command *cobra.Command) {
 	command.Flags().Int32VarP(&p.Port, "port", "p", 0, "The port where application listens on.")
 	p.markFlagMakesRevision("port")
 	command.Flags().StringArrayVarP(&p.Labels, "label", "l", []string{},
-		"Service label to set. name=value; you may provide this flag "+
+		"Labels to set for both Service and Revision. name=value; you may provide this flag "+
 			"any number of times to set multiple labels. "+
 			"To unset, specify the label name followed by a \"-\" (e.g., name-).")
 	p.markFlagMakesRevision("label")
+	command.Flags().StringArrayVarP(&p.LabelService, "label-service", "", []string{},
+		"Service label to set. name=value; you may provide this flag "+
+			"any number of times to set multiple labels. "+
+			"To unset, specify the label name followed by a \"-\" (e.g., name-). This flag takes "+
+			"precedence over \"label\" flag.")
+	p.markFlagMakesRevision("label-service")
+	command.Flags().StringArrayVarP(&p.LabelRevision, "label-revision", "", []string{},
```

Comment:
Shouldn't it be really `--label-pod` instead of `--label-revision` as it is about adding the label to the pod spec ?